src/VirtoCommerce.CustomerSegmentsModule.Core/Models/CustomerSegmentConditionPropertyValues.cs

```diff
@@ -36,5 +54,113 @@ public override bool IsSatisfiedBy(IEvaluationContext context)
 
             return result;
         }
+
+        public virtual MembersSearchCriteria BuildSearchCriteria(IMemberSearchCriteriaBuilder builder)
+        {
+            var searchPraseParts = GetDynamicPropertiesParts();
+            searchPraseParts.AddRange(GetModelPropertiesParts());
+
+            // creates a search phrase query like:
+            // "DynamicProperyName1":"DynamicProperyValue" "DynamicProperyName2":"DynamicProperyValue1","DynamicProperyValue2" "ModelPropery1":"ModelProperyValue1"
+            var searchPhrase = string.Join(' ', searchPraseParts);
+
+            builder.WithSearchPhrase(searchPhrase);
+
+            return builder.Build();
+        }
+
+        private IList<string> GetDynamicPropertiesParts()
+        {
+            var result = new List<string>();
+
+            foreach (var property in Properties)
+            {
+                if (!property.Values.IsNullOrEmpty())
+                {
+                    var propertyName = QuoteValue(property.Name);
+                    var values = property.Values.Select(x => GetSearchableValue(x.Value)).ToList();
+                    var propertyValues = string.Join(',', values);
+                    result.Add($"{propertyName}:{propertyValues}");
+                }
+            }
+
+            return result;
+        }
+
+        private IList<string> GetModelPropertiesParts()
```

Comment:
Same, can we rewrite it to pure LINQ? If not can we reduce cognitive complexity?